internal/distro/fedora30/distro_test.go

```diff
@@ -137,3 +164,136 @@ func TestImageType_BuildPackages(t *testing.T) {
 		}
 	}
 }
+
+func TestImageType_Name(t *testing.T) {
+	distro := fedora30.New()
+	imgMap := []struct {
+		arch     string
+		imgNames []string
+	}{
+		{
+			arch: "x86_64",
+			imgNames: []string{
+				"ami",
+				"ext4-filesystem",
+				"partitioned-disk",
+				"qcow2",
+				"openstack",
+				"tar",
+				"vhd",
+				"vmdk",
+			},
+		},
+		{
+			arch: "aarch64",
+			imgNames: []string{
+				"ami",
+				"ext4-filesystem",
+				"partitioned-disk",
+				"qcow2",
+				"openstack",
+				"tar",
+			},
+		},
+	}
+	for _, mapping := range imgMap {
+		arch, err := distro.GetArch(mapping.arch)
+		assert.Nil(t, err)
+		if err == nil {
```

Comment:
if you replace assert.Nil with require.Nil then the if condition isn't necessary. 

However I don't quite understand how this is going to work here. If I am on a x86_64 system and the loop iterates over the 2nd element in the array what will happen? 

What if I am on a ppc64le system ?

Reply:
All this code is independent of the architecture of the machine you are running the tests on, so you can run all the functions on any architecture.

Reply:
@atodorov if I do the replacing as you suggest, it won't be functionally equivalent IMO (failure of one scenario vs. failure of the whole test).

Reply:
I did some more changes/additions, especially I added the `TestDistro_Manifest` function I borrowed from `/internal/distro/distro_test.go` and changed a bit. I wonder if it would make sense to modify it a bit to be more generic (to include scenario file prefixes and distro objects as parameters), move it a bit higher in the tree and then call it from the more specific test files (for F30, .., RHEL-...)?